benchmark/syclblas/blas1/copy.cpp

```diff
@@ -0,0 +1,126 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename copy.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(int size, int incx, int incy) {
+  std::ostringstream str{};
+  str << "BM_Copy<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << size << "/" << incx << "/" << incy;
+  return str.str();
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, blas::SB_Handle* sb_handle_ptr, index_t size,
+         index_t incx, index_t incy, bool* success) {
+  double size_d = static_cast<double>(size);
```

Comment:
I think we can create another specialization of `blas_benchmark::utils::init_level_1_counters` for the `copy` operator and use that here instead of these counters.